web/src/components/PoiListItem.tsx

```diff
@@ -28,17 +29,20 @@ const StyledText = styled('p')({
 
 type PoiListItemProps = {
   poi: PoiModel
-  selectPoi: () => void
+  onClick?: () => void
   distance: number | null
 }
 
-const PoiListItem = ({ poi, distance, selectPoi }: PoiListItemProps): ReactElement => {
+const PoiListItem = ({ poi, distance, onClick }: PoiListItemProps): ReactElement => {
   const { t } = useTranslation('pois')
+  const [queryParams] = useSearchParams()
   const { title, category, slug } = poi
+  const slugWithQuery = `${slug}?${queryParams}`
 
   return (
     <ListItem disablePadding>
-      <StyledListItemButton onClick={selectPoi} id={slug} to={slug} component={Link} aria-label={title}>
+      {/* onClick here is just to preserve scroll position */}
```

Comment:
🔧 It doesn't really matter what `onClick` is doing here, keeping this comment means that the comment needs to be adjusted whenever some behavior is added/changed.
```suggestion
```